msteams-platform/toolkit/use-CICD-template.md

```diff
@@ -10,233 +10,368 @@ ms.date: 04/20/2022
+    :::image type="content" source="../assets/images/teams-toolkit-v2/secret.png" alt-text="Screenshot shows the keep this value secret in new variable page.":::
 
-1. Register a Microsoft Entra application in single tenant.
-2. Assign a role to your Microsoft Entra application to access your Azure subscription. The `Contributor` role is recommended.
-3. Create a new Microsoft Entra application secret.
+1. Run the pipeline.
 
-> [!TIP]
-> Save your tenant id, application id (AZURE_SERVICE_PRINCIPAL_NAME), and the secret (AZURE_SERVICE_PRINCIPAL_PASSWORD) for future use.
+    Push code to the repo to trigger pipeline.
 
-For more information, see [Azure service principals guidelines](/azure/active-directory/develop/howto-create-service-principal-portal). The following are the three ways to create service principals:
+    > [!NOTE]
+    > There's no need to commit env files under env/ folder to the repo. The env variables required for executing the CI/CD pipeline are already established in the pipeline variables.
 
-* [Microsoft Azure portal](/azure/active-directory/develop/howto-create-service-principal-portal)
-* [Windows PowerShell](/azure/active-directory/develop/howto-authenticate-service-principal-powershell)
-* [Microsoft Azure CLI](/cli/azure/create-an-azure-service-principal-azure-cli)
+    After the pipeline executes successfully, the log displays that the code is deployed to Azure and the appPackage is generated in the artifacts.
 
-## Publish Teams app using Teams Developer Portal
+    :::image type="content" source="../assets/images/teams-toolkit-v2/published.png" alt-text="Screenshot shows the pipeline runs successfully.":::
 
-If there are any changes related to Teams app's manifest file, you can update the manifest and publish the Teams app again. To publish Teams app manually, you may use [Developer Portal for Teams](https://dev.teams.microsoft.com/home).
+## Set up CI/CD pipelines using your own workflow
 
-Perform the following steps to publish your app:
+If the Teams App CLI doesn't meet your pipeline requirements, you can develop a custom deployment process that suits your needs. This section provides guidance on deploying to Azure with custom methods. If you're using a different cloud platform, refer to this section for guidance.
 
-1. Sign in to [Developer portal for Teams](https://dev.teams.microsoft.com) using the corresponding account.
-2. Import your app package in zip, select **App** > **Import app** > **Replace**.
-3. Select the target app in app list.
-4. To publish your app, select **Publish** > **Publish to your org**.
+> [!NOTE]
+> If you already have a complete CI/CD pipeline for deploying to your Azure resource, and your Teams app needs to read environment variables during runtime, configure these environment variables in the settings of your Azure resource. For testing post-deployment, see [generate the appPackage for the teams app](#generate-the-apppackage-for-the-teams-app).
```

Comment:
'For testing post-deployment' 
Suggest to change to 'post-deployment testing', because this is a way of test after the deploy.